server/src/main/java/org/jfrog/teamcity/server/runner/BaseRunTypeExtension.java

```diff
@@ -76,16 +80,36 @@ protected ModelAndView doHandle(HttpServletRequest request, HttpServletResponse
                 ModelAndView modelAndView = new ModelAndView(actualJsp);
                 modelAndView.getModel().put("controllerUrl", viewUrl);
                 modelAndView.getModel().put("deployableArtifactoryServers", deployableArtifactoryServers);
+                modelAndView.getModel().put("deployableServerIdUrlMap", deployableArtifactoryServers.getDeployableServerIdUrlMap(getProject(request)));
                 return modelAndView;
             }
         });
     }
 
+    @Nullable
+    private SProject getProject(@NotNull final HttpServletRequest request) {
```

Comment:
This methid seem to be duplicated in the code base. See the `BaseRunType` and `ArtifactoryRunTypeConfigController` classes.

Reply:
I moved the method to ServerUtils to reuse it